src/xfer/cache.rs

```diff
@@ -0,0 +1,235 @@
+use std::{
+    marker::PhantomData,
+    path::{Path, PathBuf},
+    time::{Duration, SystemTime, SystemTimeError},
+};
+
+use sha2::{Digest, Sha256};
+use snafu::{ResultExt, Snafu};
+use tokio::{fs, io};
+use url::Url;
+
+use crate::constants::DEFAULT_CACHE_MAX_AGE;
+
+pub type CacheResult<T> = Result<T, CacheError>;
+
+#[derive(Debug)]
+pub struct CacheUninitialized;
+
+#[derive(Debug)]
+pub struct CacheInitialized;
+
+#[derive(Debug, Snafu)]
+pub enum CacheError {
```

Comment:
```suggestion
pub enum Error {
```

No need to prefix everything with the module name.